pyfstat/helper_functions.py

```diff
@@ -90,24 +89,23 @@ def set_up_logger(outdir=None, label="pyfstat", log_level="INFO"):
         logger.addHandler(stream_handler)
 
     if any([type(h) == logging.FileHandler for h in logger.handlers]) is False:
-        if label:
-            if outdir:
-                if not os.path.exists(outdir):
-                    os.makedirs(outdir, exist_ok=True)
-            else:
-                outdir = "."
+        if label and outdir:
+            if not os.path.exists(outdir):
+                os.makedirs(outdir, exist_ok=True)
             log_file = os.path.join(outdir, f"{label}.log")
+            logger.info(f"Additionally logging to file: {log_file}")
             file_handler = logging.FileHandler(log_file)
             file_handler.setFormatter(common_formatter)
-
             file_handler.setLevel(level)
             logger.addHandler(file_handler)
+            # need the following because the file didn't catch any previous version logs yet
+            logger_file_only = logging.Logger("pyfstat")
+            logger_file_only.addHandler(file_handler)
+            logger_file_only.info(f"Running PyFstat version {get_version_string()}")
```

Comment:
This is ugly and I hope it doesn't lead to race conditions, but I couldn't find any documented way to issue a single line to a single handler of an existing logger. Also not sure if I should somehow close this one-off logger after use. (Unfortunately `Logger` can't be used as a context manager.)

Reply:
Why would you want this? The logger itself logs the version to all handlers.

Reply:
Because the logger can't travel back in time. If we're in this situation where there is no file handler assigned yet (`if any([type(h) == logging.FileHandler for h in logger.handlers]) is False`) but we do have `label and outdir`, then the following should have happened:
1. the user didn't pre-configure logging before `import pyfstat`
2. the fallback stdout config from `__init__.py` triggered, printing version info
3. after import, the user called `pyfstat.set_up_logger(label=label, outdir=outdir)` (as we'll want to recommend in the docs)

So at this point we'll be adding the file handler to the existing logger (which already prints to stdout), and stdout already had one round of version info. But the file didn't get it. So if we'd just raise the version info again via `logger.info` at the end of this function (deleted L109 below), it would go to both stdout and file, being duplicate in the stdout. If we did nothing, the file would have no version info. This hack makes sure it's printed once to each of the two handlers.

Alternatively we could just reinstate L109 instead, and don't worry about a duplicate version info on stdout. Or wait... I could just temporarily change levels per handler!

Reply:
Avoiding the second logger now, with same behaviour. Still hacky but less so.